packages/evm/src/interpreter.ts

```diff
@@ -441,27 +454,68 @@ export class Interpreter {
     return this._evm['_opcodeMap'][op]
   }
 
-  async _runStepHook(dynamicFee: bigint, gasLeft: bigint): Promise<void> {
-    const opcodeInfo = this.lookupOpInfo(this._runState.opCode)
-    const opcode = opcodeInfo.opcodeInfo
+  async _runStepHook(dynamicFee: bigint, gasLeft: bigint, memorySize: bigint): Promise<void> {
+    const opcodeInfo = this.lookupOpInfo(this._runState.opCode).opcodeInfo
+    const section = this._env.eof?.container.header.getSectionFromCodePosition(
+      this._runState.programCounter,
+    )
+    let error = undefined
+    let immediate = undefined
+    if (opcodeInfo.code === 0xfd) {
+      // If opcode is REVERT, read error data and return in trace
+      const [offset, length] = this._runState.stack.peek(2)
+      error = new Uint8Array(0)
+      if (length !== BIGINT_0) {
+        error = this._runState.memory.read(Number(offset), Number(length))
+      }
+    }
+
+    // Add immediate (i.e. bytecode parameter for a preceding opcode like (RJUMP 01 - jumps to PC 1))
+    const opcodesWithImmediate = [0x0e, 0xe1, 0xe2] // We exclude PUSHn because the values are on the stack (per EIP-7655)
```

Comment:
0x0e? I think 0xe0?

Take a look at `eof/stackdelta.ts` and then the intermediate fields. I suggest to read the intermediate bytes from there. There is one exception: RJUPMV, which should be handled explicitly here (there is a comment about that in there  because those intermediate bytes are dynamic, based upon the table size). There are actually much more intermediates than I expected, see for instance also EOFCREATE and RETURNCONTRACT. So if the stackdelta intermediate > 0 then report the intermediate bytes (can read directly the amount of bytes from there).

Reply:
Ah, thanks for this!  I was wondering if there something like that.